docs/messaging-implementation.md

```diff
@@ -0,0 +1,168 @@
+# In-app messaging (technical implementation)
+
+All forms of in-app messages (notifications, mascot announcements, popup announcements, etc.) share a single data model. They depend on a
+JSON feed containing certain attributes and flags. See the [configuraton](configuration.md) doc for information about how to
+point uw-frame to your desired feed.
+
+## Example message
+
+*Note: All messages must be contained within the "messages" array.*
+
+```json
+{
+    "messages": [
+        {
+            "id": 1,
+            "title": "You need to modify your NetID account to activate essential UW Services.",
+            "titleShort": null,
+            "description": "Because this message's messageType is 'notification', this field is not actually needed.",
+            "descriptionShort": null,
+            "messageType": "notification",
+            "goLiveDate": "2017-08-01T09:30",
+            "expireDate": "2017-08-02",
+            "featureImageUrl": null,
+            "priority": "high",
+            "audienceFilter": {
+                "groups": ["Users - Service Activation Required"],
+                "dataUrl": "/restProxyURL/unactivatedServices",
+                "dataObject": "services",
+                "dataArrayFilter": {"priority":"essential", "type":"netid"}
+            },
+            "actionButton": {
+                "label": "Activate services",
+                "url": "my.wisc.edu/go/example/path"
+            },
+            "moreInfoButton": null,
+            "confirmButton": null
+        }
+    ]
+}
+```
+
+**Attribute breakdown**
+
+- **id**: A unique number to identify the message. This is used for tracking seen/unseen messages, dimissed notifications, and the sort order on the notifications page.
```

Comment:
Since this is a breaking change resetting IDs anyway, should we take the opportunity to make `id` a String rather than an integer? Which would allow semantic IDs ala `fname`s.